.github/actions/diskspace-cleanup/action.yml

```diff
@@ -17,6 +17,10 @@ runs:
         set -ex
         diskspace_cutoff=${{ inputs.diskspace-cutoff }}
         docker_root_dir=$(docker info -f '{{.DockerRootDir}}')
+        if [ ! -d "$docker_root_dir" ]; then
+            echo "Docker root directory ($docker_root_dir) does not exist. Skipping disk space check."
+            exit 0
+        fi
```

Comment:
@saienduri Why does this check not work for the MI300 nodes?

Reply:
The `dind-sock` and `dind-externals` volumes are initialized as `emptyDir` for the kubernetes pods, which means they are ephemeral and do not persist any data between container restarts. So, it tries to check the Docker root directory `/var/lib/docker` which is not being explicitly mounted persistently or initialized with data, leading to the directory not existing unlike the self-hosted runner case.